plugins/modules/azure_rm_diagnosticsetting.py

```diff
@@ -0,0 +1,575 @@
+                    type: complex
+                    contains:
+                        enabled:
+                            description:
+                                - A value indicating whether the retention policy is enabled.
+                            type: bool
+                        days:
+                            description:
+                                - The number of days for the retention in days. A value of 0 will retain indefinitely.
+                            type: int
+        metrics:
+            description:
+                - The list of metric settings.
+            type: list
+            contains:
+                time_grain:
+                    description:
+                        - The timegrain of the metric in ISO8601 format.
+                    type: str
+                category:
+                    description:
+                        - Name of a Diagnostic Metric category for a resource type this setting is applied to.
+                    type: str
+                enabled:
+                    description:
+                        - A value indicating whether this metric is enabled.
+                    type: bool
+                retention_policy:
+                    description:
+                        - The retention policy for this category.
+                    type: complex
+                    contains:
+                        enabled:
+                            description:
+                                - A value indicating whether the retention policy is enabled.
+                            type: bool
+                        days:
+                            description:
+                                - The number of days for the retention in days. A value of 0 will retain indefinitely.
+                            type: int
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+import datetime
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+retention_policy_object = dict(
+    enabled=dict(type='bool', default=True, required=True),
+    days=dict(type='int', default=0, required=True)
+)
+
+log_settings_object = dict(
+    category=dict(type='str', required=True),
+    enabled=dict(type='bool', default=True, required=True),
+    retention_policy=retention_policy_object
+)
+
+metric_settings_object = dict(
+    time_grain=dict(type='str'),
+    category=dict(type='str', required=True),
+    enabled=dict(type='bool', default=True, required=True),
+    retention_policy=retention_policy_object
+)
+
+
+class AzureRMDiagnosticSetting(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str', required=True),
+            resource_id=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
```

Comment:
Argument 'state' in argument_spec defines type as 'str' but documentation doesn't define type